src/components/KMethod/KMethod.vue

```diff
@@ -0,0 +1,117 @@
+<template>
+  <KBadge
+    :background-color="backgroundColor || methodBadgeColors[method].backgroundColor"
+    class="k-method"
+    :class="`method-${method}`"
+    :color="color || methodBadgeColors[method].color"
+    max-width="auto"
+    :shape="appearance === 'rectangular' ? 'rectangular' : 'rounded'"
+  >
+    {{ methodLabel }}
+  </KBadge>
+</template>
+
+<script lang="ts">
+import { PropType, computed } from 'vue'
+import type { Method, MethodAppearance } from '@/types'
+
+export const methods: Method[] = ['get', 'post', 'put', 'patch', 'delete', 'head', 'options', 'trace', 'connect', 'custom']
+
+export const shapes: MethodAppearance[] = ['rounded', 'rectangular']
+</script>
+
+<script setup lang="ts">
+const props = defineProps({
+  method: {
+    type: String as PropType<Method>,
+    required: true,
+    validator: (value: string): boolean => {
+      return methods.includes(value as Method)
```

Comment:
suggestion: you can clean up the types here a bit
```suggestion
    validator: (value: Method): boolean => {
      return methods.includes(value)
```